packages/java/engine-runtime/src/main/java/dev/hilla/internal/ParserProcessor.java

```diff
@@ -29,6 +31,7 @@ public final class ParserProcessor {
     private String endpointAnnotationName = "dev.hilla.Endpoint";
     private String endpointExposedAnnotationName = "dev.hilla.EndpointExposed";
     private String openAPIPath;
+    private ClassLoader classLoader = ClassLoader.getSystemClassLoader();
```

Comment:
Have just checked the Jackson annotation issue: using the system class loader gets different annotation class that continues to break the Jackson parser. However, if we use the following variant:

```java
var classLoader = getClass().getClassLoader();
```

it works. 

So, I would suggest replacing this line with the following one:
```suggestion
    private ClassLoader classLoader = getClass().getClassLoader();
```